src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +61,646 @@ const ActorInit En_Rz_InitVars = {
+    diffX = curPoint->x - this->actor.world.pos.x;
+    diffZ = curPoint->z - this->actor.world.pos.z;
+    this->actor.world.rot.y = Math_Atan2S(diffX, diffZ);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 2, 0x07D0, 0xC8);
+
+    if ((SQ(diffX) + SQ(diffZ)) < SQ(10.0f)) {
+        if (this->stateFlags & EN_RZ_STATE_1) {
+            this->curPointIndex--;
+            if (this->curPointIndex < 0) {
+                return EN_RZ_PATHSTATUS_END;
+            }
+        } else {
+            this->curPointIndex++;
+            if (this->curPointIndex >= path->count) {
+                return EN_RZ_PATHSTATUS_END;
+            }
+        }
+        return EN_RZ_PATHSTATUS_AT_POINT;
+    }
+    return EN_RZ_PATHSTATUS_NORMAL;
+}
+
+s32 EnRz_CanTalk(EnRz* this, PlayState* play) {
+    if ((this->actor.xzDistToPlayer < 80.0f) && (fabsf(this->actor.playerHeightRel) < 20.0f) &&
+        Player_IsFacingActor(&this->actor, 0x3000, play)) {
+        return true;
+    }
+    return false;
+}
+
+s32 func_80BFBCEC(EnRz* this, PlayState* play) {
+    s32 pad;
+    Player* player = GET_PLAYER(play);
+
+    if ((this->actor.xzDistToPlayer < 100.0f) && Actor_IsFacingPlayer(&this->actor, 0x3000) &&
+        (player->stateFlags2 & PLAYER_STATE2_2000000)) {
+        return true;
+    }
+    return false;
+}
+
+s32 EnRz_UpdateSkelAnime(EnRz* this, PlayState* play) {
+    if (this->animIndex < EN_RZ_ANIM_LINK_NORMAL_WAIT_FREE) {
+        return SkelAnime_Update(&this->skelAnime);
+    } else {
+        return LinkAnimation_Update(play, &this->skelAnime);
+    }
+}
+
+EnRz* EnRz_FindSister(EnRz* this, PlayState* play) {
+    Actor* npc = play->actorCtx.actorLists[ACTORCAT_NPC].first;
+
+    while (npc != NULL) {
+        if ((npc->id == ACTOR_EN_RZ) && (EN_RZ_GET_TYPE(&this->actor) == EN_RZ_GET_TYPE(npc))) {
+            if (&this->actor != npc) {
+                return (EnRz*)npc;
+            }
+        }
+        npc = npc->next;
+    }
+    return NULL;
+}
+
+void func_80BFBDFC(PlayState* play) {
+    if (gSaveContext.save.weekEventReg[75] & 0x80) {
+        func_80151BB4(play, 0x27);
+    }
+    func_80151BB4(play, 0xC);
+}
+
+void EnRz_Destroy(Actor* thisx, PlayState* play) {
+    EnRz* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+s32 func_80BFBE70(EnRz* this, PlayState* play) {
+    u16 action;
+
+    if ((EN_RZ_GET_SISTER(&this->actor) == EN_RZ_JUDO) && (this->animIndex == EN_RZ_ANIM_APPLAUDING)) {
+        func_800B9010(&this->actor, NA_SE_EV_CLAPPING_2P - SFX_FLAG);
+    }
+
+    if (Cutscene_CheckActorAction(play, this->csAction)) {
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, Cutscene_GetActorActionIndex(play, this->csAction));
+        action = play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, this->csAction)]->action;
+        if (this->actionIndex != action) {
+            this->actionIndex = action;
+            switch (action & 0xFFFF) {
```

Comment:
```suggestion
            switch (this->actionIndex) {
```
Bump, this should match according to Elliptic
matches: https://decomp.me/scratch/Tn9v8

Reply:
Oh yeah, I missed that not being addressed.

Reply:
Oh yeah, I missed that not being addressed.

Reply:
up